src/extensions/scratch3_mqtt/index.js

```diff
@@ -303,60 +302,48 @@ class Scratch3MQTTBlocks {
 
     disconnectFromBroker(args) {
         console.log("[mqtt] disconnecting from MQTT broker... ");
-        if ( this._client != null ) {  this._client.end(); }
+        if ( this._client ) {  this._client.end(); }
```

Comment:
Using `truthy` and `falsey` comparison instead of comparing with `null` or `undefined` values can eliminate potential bugs; because unfortunately the javascript equality matrix is pretty complicated https://dorey.github.io/JavaScript-Equality-Table/unified/ 😢